packages/nestjs-redis/src/redis.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Inject, Injectable, OnModuleDestroy, Provider } from '@nestjs/common';
+import IORedis from 'ioredis';
+
+import { Providers } from './constants';
+import { RedisConnection, RedisOptions } from './interfaces/options.interface';
+
+@Injectable()
+export class RedisService implements OnModuleDestroy {
+  static connectionFactory: Provider<Promise<RedisConnection>> = {
+    inject: [
+      {
+        token: Providers.REDIS_OPTIONS,
+        optional: false,
+      },
+    ],
+    provide: Providers.REDIS_CONNECTION,
+    useFactory: async (config: RedisOptions): Promise<RedisConnection> => {
+      return new IORedis(config.port, config.host, config.ioredis);
+    },
```

Comment:
thanks for supporting the additional `config.ioredis` options! saves us a bit of headache in the future if we end up needing it